go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/kong/kubernetes-testing-framework
 
-go 1.19
+go 1.21
+
+toolchain go1.21.0
```

Comment:
While I understand the motivation I also see this as an end user facing change because this will bump  the minimum required version to compile ktf (and modules that require it) 🤔 

OTOH though we're still pre v1 with ktf so I guess we should be fine.